src/agents/workflows/subgraphs/step4_features/solve/graph.py

```
@@ -200,26 +216,45 @@ def execute_strategy(state: AgentState) -> AgentState:
+        allow_local_fallback = _env_truthy("SOLVE_LOCAL_PYTHON_FALLBACK", "1")
+        if allow_local_fallback:
+            execution_backend = "local"
+            local_ok, local_stdout, local_stderr, local_text, local_error = (
+                _run_python_locally(code)
+            )
+            success = local_ok
+            stdout = local_stdout
+            stderr = list(local_stderr)
+            if local_error:
+                stderr.append(f"Local execution error: {local_error}")
+            stderr.append(f"E2B fallback reason: {str(exc)}")
```

Ruff (0.15.0)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)